contracts/Vault.vy

```diff
@@ -142,57 +204,169 @@ def setSymbol(_symbol: String[20]):
+        Changes the rewards address. Any distributed rewards
+        will cease flowing to the old address and begin flowing
+        to this address once the change is in effect.
+
+        This will not change any transactions in progress, only
+        new transactions made after this change goes into effect.
+
+        This may only be called by governance.
+    @param _rewards The address to use for collecting rewards.
+    """
     assert msg.sender == self.governance
     self.rewards = _rewards
 
 
 @external
 def setDepositLimit(_limit: uint256):
+     """
+    @notice
+        Changes the maximum amount of tokens a single address may have
+        deposited in this Vault.
+
+        Note, this is not how much may be deposited in a single transaction,
+        but the maximum amount that may be deposited across all transactions.
+
+        This may only be called by governance.
+    @param _limit The new deposit limit to use.
+    """
     assert msg.sender == self.governance
     self.depositLimit = _limit
 
 
 @external
 def setPerformanceFee(_fee: uint256):
+    """
+    @notice
+        Used to change the value of `performanceFee`.
+
+        This may only be called by governance.
+    @param _fee The new performance fee to use.
+    """
     assert msg.sender == self.governance
     self.performanceFee = _fee
 
 
 @external
 def setManagementFee(_fee: uint256):
+    """
+    @notice
+        Used to change the value of `managementFee`.
+
+        This may only be called by governance.
+    @param _fee The new management fee to use.
+    """
     assert msg.sender == self.governance
     self.managementFee = _fee
 
 
 @external
 def setGuardian(_guardian: address):
+    """
+    @notice
+        Used to change the address of `guardian`.
+
+        This may only be called by governance or the existing guardian.
+    @param _guardian The new guardian address to use.
+    """
     assert msg.sender in [self.guardian, self.governance]
     self.guardian = _guardian
 
 
 @external
 def setEmergencyShutdown(_active: bool):
     """
-    Activates Vault mode where all Strategies go into full withdrawal
+    @notice
+        Activates or deactivates Vault mode where all Strategies go into full
+        withdrawal.
+
+        During Emergency Shutdown mode of the Vault, the intention is for the
+        Vault to recall the debt as quickly as reasonable (given on-chain
+        conditions) with minimal losses, and open up deposits to be withdrawn
+        by Users as easily as possible. There is no restrictions on
+        withdrawals above what is expected under Normal Operation.
+
+        1. During Emergency Shutdown, no Users may deposit into the Vault
+        2. During Emergency Shutdown, Governance cannot add new Strategies
+        3. During Emergency Shutdown, each Strategy must pay back their debt as
+            quickly as reasonable to minimally affect their position
+        4. Only Governance can undo Emergency Shutdown
```

Comment:
We already have this at the contract level, could it perhaps be possible to simply reference back to that?

Reply:
Certainly, I'll rework it.